src/core/lib/security/credentials/tls/tls_credentials.cc

```diff
@@ -46,6 +46,12 @@ bool CredentialOptionSanityCheck(grpc_tls_credentials_options* options,
     gpr_log(GPR_ERROR, "TLS credentials options is nullptr.");
     return false;
   }
+  // In this case, there will be non-retriable handshake errors.
+  if (options->min_tls_version() == grpc_tls_version::TLS1_3 &&
+      options->max_tls_version() == grpc_tls_version::TLS1_2) {
+    gpr_log(GPR_ERROR, "TLS min version must not be higher than max version.");
```

Comment:
Should we use a bit more informative error message here? E.g. if min or max < 1.2, can say the min is too low? if max or min > 1.3, say not supported? and then also have a min <= max check?

Reply:
Added those checks and tests. PTAL